arbitrum-client/src/client/mod.rs

```diff
@@ -1,29 +1,36 @@
 //! The definition of the Arbitrum client, which holds the configuration
 //! details, along with a lower-level handle for the darkpool smart contract
 
-use std::{str::FromStr, sync::Arc, time::Duration};
+use std::str::FromStr;
 
-use alloy_primitives::{Address as AlloyAddress, ChainId};
-use constants::{DEVNET_DEPLOY_BLOCK, MAINNET_DEPLOY_BLOCK, TESTNET_DEPLOY_BLOCK};
-use ethers::{
-    core::k256::ecdsa::SigningKey,
-    middleware::{MiddlewareBuilder, NonceManagerMiddleware, SignerMiddleware},
-    providers::{Http, Middleware, Provider},
-    signers::{LocalWallet, Signer, Wallet},
-    types::{Address, BlockNumber},
+use alloy::{
+    providers::{DynProvider, Provider, ProviderBuilder},
+    signers::local::PrivateKeySigner,
+    transports::http::reqwest::Url,
 };
+use alloy_contract::CallBuilder;
+use alloy_primitives::{Address, BlockNumber, ChainId};
+use constants::{DEVNET_DEPLOY_BLOCK, MAINNET_DEPLOY_BLOCK, TESTNET_DEPLOY_BLOCK};
 use util::err_str;
 
 use crate::{
-    abi::DarkpoolContract,
+    abi::Darkpool::DarkpoolInstance,
     constants::Chain,
     errors::{ArbitrumClientConfigError, ArbitrumClientError},
 };
 
 mod contract_interaction;
 mod event_indexing;
 
-/// A configuration struct for the Arbitrum client, consists of relevant
```

Comment:
nit: looks like this got deleted, should add it back to the top of the `ArbitrumClientConfig` doc comment